### PR TITLE
Implemented click

### PR DESCRIPTION
**Pull request by retr0h**
First round conversion from docopt to click.

* No longer using docopt and the crude leveraging of subcommands.
  The result is less-meta, but more readable.
* Reworked the molecule constructor to accept args and subcommand
  args.  Previously these were munged together.
* Updated changelog as to not forget the compatibility issues.
* Removed CLI test.
* Removed vagrant command create unit test.  This is handled with
  the functional tests.  Still need to continue implementing
  remaining driver agnostic command tests.

Fixes: #461

### CODE REVIEW

**Comment by signed8bit** [comment]
I want to review this right now but this is a big one. Give me some play here. :-)

**Comment by retr0h** [comment]
```
  syntax: commands succeeded
  py27-ansible19-unit: commands succeeded
  py27-ansible19-functional: commands succeeded
  py27-ansible20-unit: commands succeeded
  py27-ansible20-functional: commands succeeded
  py27-ansible21-unit: commands succeeded
  py27-ansible21-functional: commands succeeded
  doc: commands succeeded
  congratulations :)
```

File: molecule/cli.py [hunk 1]
```diff
@@ -17,56 +17,36 @@
...
-
-
-def main():
-    CLI().main()
 
 
-if __name__ == '__main__':
-    main()
+@click.group()
+@click.option('--debug/--no-debug', default=False)
```
Comment (signed8bit) [inline molecule/cli.py]: Is there ever a chance where debug would be the default? I see how the Click docs recommend you provide an enable/disable but in the case of debug it feels unnecessary. Without knowing the default, I'd wonder if users would be compelled to use --no-debug needlessly?

Comment (retr0h) [inline molecule/cli.py]: Yeah, I don't know.  I kinda went along with the click "way".  You can't use a boolean flag w/o proving a way to turn the flag off.

Debug currently doesn't do anything in molecule, other than displaying output of a couple subcommands.  I felt I should just do the click way.

**Comment by retr0h** [comment]
```
  syntax: commands succeeded
  py27-ansible19-unit: commands succeeded
  py27-ansible19-functional: commands succeeded
  py27-ansible20-unit: commands succeeded
  py27-ansible20-functional: commands succeeded
  py27-ansible21-unit: commands succeeded
  py27-ansible21-functional: commands succeeded
  doc: commands succeeded
  congratulations :)
```

File: molecule/cli.py (hunk 1 above)
Comment (signed8bit) [inline molecule/cli.py]: Perhaps add a help message to the debug option to clarify the default...

    help='Enable or disable debug mode. Default is disabled.' 

something or other.

Comment (retr0h) [inline molecule/cli.py]: > Perhaps add a help message to the debug option to clarify the default...

Good call.

**Comment by signed8bit** [comment]
👍

**Pull request merged**